File: clients/java/src/main/java/com/blazecache/TestSinglePut.java
package com.blazecache;

import java.util.Arrays;
import java.util.List;

public class TestSinglePut {
    public static void main(String[] args) {
        List<String> servers = Arrays.asList(
            "127.0.0.1:6784",
            "127.0.0.1:6786",
            "127.0.0.1:6788"
        );

        System.out.println("Creating client...");
        CacheClient client = new CacheClient(servers, CacheClient.SelectionStrategy.CONSISTENT_HASHING);

        try {
            System.out.println("Testing ping...");
            client.ping();
            System.out.println("✓ Ping successful");

            System.out.println("Testing PUT...");
            client.set("test_key_1", "test_value_1".getBytes());
            System.out.println("✓ PUT successful");

            System.out.println("Testing GET...");
            java.util.Optional<byte[]> result = client.get("test_key_1");
            if (result.isPresent()) {
                System.out.println("✓ GET successful: key=test_key_1, value=" + new String(result.get()));
            } else {
                System.out.println("✗ GET returned empty");
            }

            System.out.println("\n✅ Single PUT test completed successfully!");
        } catch (Exception e) {
            System.err.println("❌ Test failed: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}

File: clients/go/examples/test_single_put.go
package main

import (
	"fmt"
	"log"

	blazecache "github.com/blazecache/client"
)

func main() {
	servers := []string{
		"127.0.0.1:6784",
		"127.0.0.1:6786",
		"127.0.0.1:6788",
	}

	fmt.Println("Creating client...")
	c, err := blazecache.New(servers...)
	if err != nil {
		log.Fatalf("Failed to create client: %v", err)
	}

	c = c.WithStrategy(blazecache.ConsistentHashing)

	fmt.Println("Testing ping...")
	if err := c.Ping(); err != nil {
		log.Fatalf("Ping failed: %v", err)
	}
	fmt.Println("✓ Ping successful")

	fmt.Println("Testing PUT...")
	item := &blazecache.Item{
		Key:   "test_key_1",
		Value: []byte("test_value_1"),
	}
	if err := c.Set(item); err != nil {
		log.Fatalf("PUT failed: %v", err)
	}
	fmt.Println("✓ PUT successful")

	fmt.Println("Testing GET...")
	result, err := c.Get("test_key_1")
	if err != nil {
		log.Fatalf("GET failed: %v", err)
	}
	if result == nil {
		log.Fatalf("GET returned nil")
	}
	fmt.Printf("✓ GET successful: key=%s, value=%s\n", result.Key, string(result.Value))

	fmt.Println("\n✅ Single PUT test completed successfully!")
}
